scripts/prettier.js

```diff
@@ -0,0 +1,67 @@
+/* eslint-disable no-console */
+
+// Based on similar script in React
+// https://github.com/facebook/react/blob/b87aabdfe1b7461e7331abb3601d9e6bb27544bc/scripts/prettier/index.js
+
+import glob from 'glob';
+import prettier from 'prettier';
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+import listChangedFiles from './listChangedFiles';
+
+const prettierConfigPath = path.join(__dirname, '../prettier.config.js');
+
+const mode = process.argv[2] || 'check';
+const shouldWrite = mode === 'write' || mode === 'write-changed';
+const onlyChanged = mode === 'check-changed' || mode === 'write-changed';
+
+const changedFiles = onlyChanged ? listChangedFiles() : null;
+let didWarn = false;
+let didError = false;
+
+const ignoredFiles = fs
+  .readFileSync('.eslintignore', 'utf-8')
+  .split(os.EOL)
+  .filter(notEmpty => notEmpty)
+  .map(file => `**/${file}/**`);
```

Comment:
Glob doesn't work with file paths but I've made the code a bit smarter here it checks to see if it needs to add anything rather than just blanket appending a prefix and suffix

Reply:
According to https://eslint.org/docs/user-guide/configuring#ignoring-files-and-directories it is using `node-ignore`. I think we should use that also to be consistent